spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/job/support/StepExecutionResourceBuilder.java

```diff
@@ -34,7 +34,7 @@
 public class StepExecutionResourceBuilder {
 
 	static public StepExecutionResource toResource(StepExecution entity) {
-		return new StepExecutionResource(entity.getId(), entity, generateStepType(entity));
+		return new StepExecutionResource(entity.getJobExecution().getId(), entity, generateStepType(entity));
```

Comment:
Remind me, why `id` for step execution should be one from job execution and not from its own entity?